test/NVQPP/if_jit.cpp

```diff
@@ -0,0 +1,26 @@
+/*******************************************************************************
+ * Copyright (c) 2022 - 2023 NVIDIA Corporation & Affiliates.                  *
+ * All rights reserved.                                                        *
+ *                                                                             *
+ * This source code and the accompanying materials are made available under    *
+ * the terms of the Apache License 2.0 which accompanies this distribution.    *
+ ******************************************************************************/
+
+// RUN: nvq++ %s --target quantinuum --emulate && ./a.out | FileCheck %s
```

Comment:
If we have multiple tests that compile without setting `-o <something>`, couldn't we have conflicts with the `a.out` name ?

```suggestion
// RUN: nvq++ %s -o %basename_t.x --target quantinuum --emulate && ./%basename_t.x | FileCheck %s
```

Reply:
In my build I got all in the same directory:
```
build/test/NVQPP/a.out
build/test/NVQPP/include.x
build/test/NVQPP/introspection.x
build/test/NVQPP/out_auto_kernel.x
build/test/NVQPP/out_testdcl.x
build/test/NVQPP/out_testifstmts.x
build/test/NVQPP/out_testlambdaI.x
build/test/NVQPP/out_testTemplateI.x
```
There is already a `a.out` from `test/NVQPP/bug_qubit.cpp`

Reply:
Ok. Thanks.